botorch/optim/optimize_mixed.py

```diff
@@ -903,7 +983,7 @@ def optimize_acqf_mixed_alternating(
             "integers between 0 and num_dims - 1."
         )
     discrete_dims_t = torch.tensor(
-        discrete_dims, dtype=torch.long, device=tkwargs["device"]
+        list(discrete_dims.keys()), dtype=torch.long, device=tkwargs["device"]
     )
```

Comment:
This seems unused after your changes

Reply:
I thought so as well at first but it's used in l 1020